uberfire-commons-editor/uberfire-commons-editor-client/src/main/java/org/uberfire/ext/editor/commons/client/file/CopyPopupView.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2015 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.uberfire.ext.editor.commons.client.file;
+
+import com.github.gwtbootstrap.client.ui.Button;
+import com.github.gwtbootstrap.client.ui.TextBox;
+import com.github.gwtbootstrap.client.ui.constants.ButtonType;
+import com.github.gwtbootstrap.client.ui.constants.IconType;
+import com.google.gwt.event.dom.client.HasClickHandlers;
+import com.google.gwt.user.client.Window;
+import com.google.gwt.user.client.ui.HasText;
+import org.uberfire.ext.editor.commons.client.resources.CommonImages;
+import org.uberfire.ext.editor.commons.client.resources.i18n.CommonConstants;
+import org.uberfire.ext.widgets.common.client.common.popups.FormStylePopup;
+import org.uberfire.ext.widgets.common.client.common.popups.footers.GenericModalFooter;
+
+public class CopyPopupView extends FormStylePopup implements CopyPopup.View {
+
+    final private TextBox nameTextBox = new TextBox();
+    final private TextBox checkInCommentTextBox = new TextBox();
+    private final Button copy;
+    private final Button cancel;
+
+    public CopyPopupView() {
+        super( CommonImages.INSTANCE.edit(),
+               CommonConstants.INSTANCE.CopyPopupTitle() );
+        //Make sure it appears on top of other popups
+        getElement().getStyle().setZIndex( Integer.MAX_VALUE );
+
+        nameTextBox.setTitle( CommonConstants.INSTANCE.NewName() );
+        nameTextBox.setWidth( "200px" );
+        addAttribute( CommonConstants.INSTANCE.NewNameColon(),
+                      nameTextBox );
+
+        checkInCommentTextBox.setTitle( CommonConstants.INSTANCE.CheckInComment() );
+        checkInCommentTextBox.setWidth( "200px" );
+        addAttribute( CommonConstants.INSTANCE.CheckInCommentColon(),
+                      checkInCommentTextBox );
+        hide();
+
+        GenericModalFooter footer = new GenericModalFooter();
+        this.copy = footer.addButton( CommonConstants.INSTANCE.CopyPopupCreateACopy(),
+                                      IconType.SAVE,
+                                      ButtonType.PRIMARY );
```

Comment:
What I do is. I actually do keep the command in the view, but then pass the action to the presenter with a method that has a name that describes the action. This way the presenter does not care about click handlers and since the method name in the presenter interface describes the action that is taken, the test is more readable.

Just thought I mention what I do. Any client side code that is testable with MVP warms my heart ;)


Example:
https://github.com/droolsjbpm/drools-wb/blob/master/drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/wizard/pages/SummaryPageView.java#L32-32